internal/engine/redis/redis.go

```
@@ -819,6 +830,58 @@ func (r *Client) DeleteIfEqual(ctx context.Context, key, expectedValue string) b
 }
 
+// AcquireLease atomically creates an owner-scoped lease. A false result with
+// no error means another owner currently holds the lease.
+func (r *Client) AcquireLease(ctx context.Context, key, owner string, ttl time.Duration) (bool, error) {
+	if r == nil || r.client == nil {
+		return false, fmt.Errorf("redis: not initialised")
+	}
+	if ttl <= 0 {
+		return false, fmt.Errorf("redis: lease TTL must be positive")
+	}
+	ok, err := r.client.SetNX(ctx, key, owner, ttl).Result()
```

golangci-lint (2.12.2)

[error] SA1019: r.client.SetNX is deprecated: Use Set with NX option instead as of Redis 2.6.12.

(staticcheck)